ospd_openvas/preferencehandler.py

```diff
@@ -0,0 +1,610 @@
+                )
+                cred_prefs_list.append(
+                    OID_SMB_AUTH
+                    + ':2:'
+                    + 'password:SMB password:|||'
+                    + '{0}'.format(password)
+                )
+            if service == 'esxi':
+                cred_prefs_list.append(
+                    OID_ESXI_AUTH
+                    + ':1:entry:'
+                    + 'ESXi login name:|||'
+                    + '{0}'.format(username)
+                )
+                cred_prefs_list.append(
+                    OID_ESXI_AUTH
+                    + ':2:'
+                    + 'password:ESXi login password:|||'
+                    + '{0}'.format(password)
+                )
+
+            if service == 'snmp':
+                community = cred_params.get('community', '')
+                auth_algorithm = cred_params.get('auth_algorithm', '')
+                privacy_password = cred_params.get('privacy_password', '')
+                privacy_algorithm = cred_params.get('privacy_algorithm', '')
+
+                cred_prefs_list.append(
+                    OID_SNMP_AUTH
+                    + ':1:'
+                    + 'password:SNMP Community:|||'
+                    + '{0}'.format(community)
+                )
+                cred_prefs_list.append(
+                    OID_SNMP_AUTH
+                    + ':2:'
+                    + 'entry:SNMPv3 Username:|||'
+                    + '{0}'.format(username)
+                )
+                cred_prefs_list.append(
+                    OID_SNMP_AUTH + ':3:'
+                    'password:SNMPv3 Password:|||' + '{0}'.format(password)
+                )
+                cred_prefs_list.append(
+                    OID_SNMP_AUTH
+                    + ':4:'
+                    + 'radio:SNMPv3 Authentication Algorithm:|||'
+                    + '{0}'.format(auth_algorithm)
+                )
+                cred_prefs_list.append(
+                    OID_SNMP_AUTH
+                    + ':5:'
+                    + 'password:SNMPv3 Privacy Password:|||'
+                    + '{0}'.format(privacy_password)
+                )
+                cred_prefs_list.append(
+                    OID_SNMP_AUTH
+                    + ':6:'
+                    + 'radio:SNMPv3 Privacy Algorithm:|||'
+                    + '{0}'.format(privacy_algorithm)
+                )
+
+        return cred_prefs_list
+
+    def set_credentials(self) -> bool:
+        """ Get the credentials from the scan collection and store them
+        in the kb. """
+        credentials = self.scan_collection.get_credentials(self.scan_id)
+        if credentials:
+            cred_prefs = self.build_credentials_as_prefs(credentials)
+            if cred_prefs:
+                self.kbdb.add_scan_preferences(self.openvas_scan_id, cred_prefs)
+                return True
+
+        return False
+
+    def set_main_kbindex(self, main_kbindex: int):
```

Comment:
It the main_kbindex really necessary? I think it is the same as self.kbdb.index.

Reply:
Yes, we need it on the OpenVAS side.

Reply:
I wanted to express if passing the main_kbindex is necessary here. imho we already have the information in self.kbdb. I think the passed main_kbindex is the same as self.kbdb here.